modules/puppetca/puppetca_main.rb

```diff
@@ -47,11 +46,12 @@ def disable certname
     def autosign certname
       FileUtils.touch(autosign_file) unless File.exist?(autosign_file)
 
-      autosign = File.open(autosign_file, File::RDWR)
-      # Check that we don't have that host already
-      found = autosign.readlines.find { |line| line.chomp == certname }
-      autosign.puts certname unless found
-      autosign.close
+      open(autosign_file, File::RDWR) do |autosign|
+        # Check that we don't have that host already
+        found = false
+        autosign.each_line { |line| found = true if line.chomp == certname }
```

Comment:
@ekohl: On further thought I believe that you are right and that the original 

found = autosign.readlines.find { |line| line.chomp == certname }

is better than the 

autosign.each_line { |line| found = true if line.chomp == certname }

You are welcome to fix this if you wish. Unfortunately I do not have the time to test this now. When I get some space then I will switch it back.